ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/resources/arjunacore/XAResourceRecord.java

```diff
@@ -297,13 +297,17 @@ public int topLevelAbort()
 						&& (e1.errorCode < XAException.XA_RBEND))
 				    {
 					/*
-					 * Has been marked as rollback-only. We still
-					 * need to call rollback.
+					 * Has been marked as rollback-only. If the branch has not been
+					 * already rolled back (i.e. _rolledBack == true ), we still need
```

Comment:
I think it might be clearer to give the code example in the inverse of how it is here, in other words `(i.e. _rolledBack = false)` - please let me know what you think? The text before and after the example appear to be describing that case.

Reply:
Done, thanks @tomjenkinson